AutoGear.lua

```diff
@@ -1889,80 +1891,99 @@ AutoGearFrame:SetScript("OnEvent", function (this, event, arg1, arg2, arg3, arg4
 		elseif (event == "QUEST_DETAIL") then
 			QuestDetailAcceptButton_OnClick()
 		elseif (event == "GOSSIP_SHOW") then
-			--active quests
-			if (not IsClassic) then
-				local quests = C_GossipInfo.GetNumActiveQuests()
-				local info = {C_GossipInfo.GetActiveQuests()}
-				for i = 1, quests do
-					local title, questLevel, isTrivial, frequency, repeatable, isComplete, isLegendary = info[(i-1)*6+1], info[(i-1)*6+2], info[(i-1)*6+3], info[(i-1)*6+4], info[(i-1)*6+5], info[(i-1)*6+6], info[(i-1)*6+7]
-					if (isComplete) then
+			--active quests			
+			if (SL) then
+				for i = 1, C_GossipInfo.GetNumActiveQuests() do
+					local quest = C_GossipInfo.GetActiveQuests()[i]
+					if (quest["isComplete"]==true) then
 						C_GossipInfo.SelectActiveQuest(i)
 					end
 				end
-			else
-				local quests = GetNumGossipActiveQuests()
+			end
```

Comment:
Even though I realize it's not strictly necessary because the values are mutually exclusive, please use ```elseif``` here instead of end and if to clarify exclusivity.

I realize the code is really messy already, but I'm just trying to keep everything I put in it from now on as clean as possible.  AutoGear was first written by my brother in 2009 when he was less experienced with code cleanliness.